selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaTerminal.java

```diff
@@ -59,20 +62,82 @@ private TheiaTerminal(
 
   public interface Locators {
     String TERMINAL_BODY_ID_TEMPLATE = "terminal-%s";
-    String TERMINAL_TEXT_LAYER = "//canvas[@class='xterm-text-layer']";
-    String TERMINAL_CURSOR_LAYER = "//canvas[@class='xterm-cursor-layer']";
+    String TERMINAL_TEXT_LAYER = "(//canvas[@class='xterm-text-layer'])[%s]";
+    String TERMINAL_CURSOR_LAYER = "(//canvas[@class='xterm-cursor-layer'])[%s]";
+    String TERMINAL_TAB_XPATH_TEMPLATE = "//div[@class='p-TabBar-tabLabel' and text()='%s']";
   }
 
-  private WebElement getTerminalTextLayer() {
-    return seleniumWebDriverHelper.waitVisibility(By.xpath(TERMINAL_TEXT_LAYER));
+  private String getTerminalCursorLayer(int terminalIndex) {
+    final int adaptedTerminalIndex = terminalIndex + 1;
+
+    return format(TERMINAL_CURSOR_LAYER, adaptedTerminalIndex);
+  }
+
+  private String getTerminalTextLayerXpath(int terminalIndex) {
+    final int adaptedTerminalIndex = terminalIndex + 1;
+
+    return format(TERMINAL_TEXT_LAYER, adaptedTerminalIndex);
+  }
+
+  private String getTerminalTabXpath(String tabTitle) {
+    return format(TERMINAL_TAB_XPATH_TEMPLATE, tabTitle);
+  }
+
+  private WebElement getTerminalTextLayer(int terminalIndex) {
+    final String terminalTextLayerXpath = getTerminalTextLayerXpath(terminalIndex);
+
+    return seleniumWebDriverHelper.waitVisibility(By.xpath(terminalTextLayerXpath));
+  }
+
+  public void waitTabSelected(String tabTitle) {
```

Comment:
I see there are two patterns of method names includes`Terminal`. It looks redundant so as it is about Theia Terminal anyway.